text/0224-experimental-semantic-conventions.md

```diff
@@ -0,0 +1,115 @@
+# Allowing experimental extensions to Semantic Conventions
+
+A process that allows experimental / non-stable semantic conventions to exist within stable versions and how they evolve into stable semantic conventions.
+
+## Motivation
+
+Today, semantic conventions are released as a large bundle of "all or nothing".
+OpenTelemetry struggles with ensuring semantic conventions represent true "generic" need in observability in addition to providing a mechanism to "discover" or "experiment" with a standard.
+This [Slack thread on #otel-specification](https://cloud-native.slack.com/archives/C01N7PP1THC/p1667997213067169?thread_ts=1667933561.313819&cid=C01N7PP1THC) highlights the problem in context of a specific semantic convention pull request.
+What should OpenTelemetry do with semantic convention proposals when there's unclear evidence that a particular signal is necessary or ubiquitous?
+
+## Explanation
+
+This proposal generally follows the recommendations of [RFC6648 section 4 “Recommendations for Protocol Designers.”](https://www.rfc-editor.org/rfc/rfc6648#section-4)
+RFC6648 recommends a registry of names be established which standardizes usage and prevents collisions.
+We already have such a registry in the form of our semantic conventions.
+RFC6648 further recommends that X- prefix (or similar structure) MUST NOT be defined as experimental or nonstandard, and that names missing the X- prefix MUST NOT be defined as stable or standard.
+
+## Definitions
+
+**Attribute**: For the purposes of this document, an attribute may refer to any component of any signal emitted by OpenTelemetry instrumentation.
+This may include span or metric attribute keys and values, span or metric names, log or event attribute names and values, or any other value emitted by OTel instrumentation.
+
+**Private Attribute**: An attribute is private if it is meant only for the exclusive private use of the instrumentation author, and is not expected to be emitted by any public library or understood by any public backend tools.
+
+**Registered Attribute**: A registered attribute is any attribute which is in the provisional registry outlined in the proposal or the semantic conventions.
+
+## Proposal
+
+For the purposes of this document only attributes are discussed, but it should be understood that the TC may decide to make other telemetry components such as metric names, event names, or other components registerable.
```

Comment:
nit: relationships between spans

Reply:
Limited scope of the proposal to an attribute dictionary